appengine/genetics/js/blocks.js

```diff
@@ -0,0 +1,465 @@
+        'MATE_ANSWER': BlocklyGames.getMsg('Genetics_mateAnswerOwnerTooltip')
+      };
+      return TOOLTIPS[mode];
+    });
+  },
+  /**
+   * Modify this block to have the correct output type.
+   * @param {string} newProp Either a property requesting a function owner that
+   * returns a string or a property that returns a number.
+   * @private
+   * @this Blockly.Block
+   */
+  updateType_: function(newProp) {
+    if (newProp == 'SEX' || newProp == 'PICK_FIGHT' ||
+        newProp == 'CHOOSE_MATE' || newProp == 'MATE_ANSWER') {
+      this.outputConnection.setCheck('String');
+    } else {
+      this.outputConnection.setCheck('Number');
+    }
+  }
+};
+
+/**
+ * Defines the JavaScript generation for getProperties.
+ * @param {Blockly.Block} block
+ * @return {!Array.<string|number>}
+ */
+Blockly.JavaScript['genetics_getProperties'] = function(block) {
+  // Generate JavaScript for getting mouse property.
+  var mouse = Blockly.JavaScript.valueToCode(block, 'MOUSE',
+          Blockly.JavaScript.ORDER_NONE) || 'me()';
+  var property = block.getFieldValue('PROPERTY');
+  var code = mouse + '.';
+  switch (property) {
+    case 'SIZE':
+      code += 'size';
+      break;
+    case 'AGGRESSIVENESS':
+      code += 'aggressiveness';
+      break;
+    case 'START_AGGRESSIVENESS':
+      code += 'aggressiveness';
+      break;
+    case 'FERTILITY':
+      code += 'fertility';
+      break;
+    case 'START_FERTILITY':
+      code += 'startFertility';
+      break;
+    case 'SEX':
+      code += 'sex';
+      break;
+    case 'AGE':
+      code += 'age';
+      break;
+    case 'ID':
+      code += 'id';
+      break;
+    case 'PICK_FIGHT':
+      code += 'pickFightOwner';
+      break;
+    case 'CHOOSE_MATE':
+      code += 'chooseMateOwner';
+      break;
+    case 'MATE_ANSWER':
+      code += 'mateAnswerOwner';
+      break;
+    default:
+      throw 'Unknown math operator: ' + operator;
+  }
+  return [code, Blockly.JavaScript.ORDER_MEMBER];
+};
+
+/**
+ * Block for getting the enum for mouse sexes.
+ * @this Blockly.Block
+ */
+Blockly.Blocks['genetics_sex'] = {
+  /**
+   * Block for getting mouse sexes
+   * @this Blockly.Block
+   */
+  init: function() {
+    this.jsonInit({
+      "message0": "Sex.%1",
+      "args0": [
+        {
+          "type": "field_dropdown",
+          "name": "PROPERTY",
+          "options": [
+            ['Hermaphrodite', 'HERMAPHRODITE'],
+            ['Male', 'MALE'],
+            ['Female', 'FEMALE']
+          ]
+        }
+      ],
+      "output": "String",
+      "colour": Genetics.Blocks.GENETICS_HUE
+    });
+  }
+};
+
+/**
+ * Defines the JavaScript generation for sex.
+ * @param {Blockly.Block} block
+ * @return {!Array.<string|number>}
+ */
+Blockly.JavaScript['genetics_sex'] = function(block) {
+  // Generate JavaScript for getting sex enum.
+  var property = block.getFieldValue('PROPERTY');
+  var code = mouse + 'Sex.';
+  switch (property) {
```

Comment:
I'm not worried about length; it's more the duplication that's the issue (though I take your point that it's less likely to be changed). That said, it might be quite easy to do since you already do have the object defined in Genetics.Mouse.Sex.